app/models/nonprofit.rb

```diff
@@ -276,6 +277,13 @@ def add_billing_subscription
     b_sub = build_billing_subscription(billing_plan: billing_plan, status: 'active')
   end
 
+  def add_scheme_to_website
+    website = self.website
+    if website.present? && URI.parse(website).scheme.blank?
```

Comment:
If you pass in a totally invalid URI, a URI::InvalidURIError will be raised. Like what happens if they pass an array? or an invalid string? Or a hash? 

The validation should be catching bad URL's. You may want to refer to or reuse the code in `validate_url`'s source to reduce the chance of forgetting something. (Although totally possible they haven't thought of everything as well.)

Reply:
I used just some parts of it regarding the scheme to do nothing if it is valid, the rest would be duplicated with the validation itself. What do you think?